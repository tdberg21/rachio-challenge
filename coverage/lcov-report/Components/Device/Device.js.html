<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Components/Device/Device.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">Components/Device</a> Device.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>13/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">96.15% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>25/26</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import ZoneContainer from '../ZoneContainer/ZoneContainer.js';
import { apiKey } from '../../helpers/apiKey.js';
&nbsp;
export default class Device extends Component {
  constructor() {
    super();
    
    this.state = {
      zones: [],
      duration: 0,
      status: []
    };
    this.scrubZones = this.scrubZones.bind(this);
    this.handleDisplayZones = this.handleDisplayZones.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.activateAllZones = this.activateAllZones.bind(this);
    this.activateZone = this.activateZone.bind(this);
  }
&nbsp;
  scrubZones(zone) {
    let {
      id, name, zoneNumber, enabled, lastWateredDate, maxRuntime, imageUrl
    } = zone;
    let zoneInfo = {
      id, name, zoneNumber, enabled, lastWateredDate, maxRuntime, imageUrl
    };
    return zoneInfo;
  }
&nbsp;
  handleDisplayZones(zones) {
    if (this.state.zones.length) {
      this.setState({
        zones: []
      });
&nbsp;
    } else {
      let cleanZones = zones.map(zone =&gt; this.scrubZones(zone));
      this.setState({
        zones: cleanZones
      });
    }
  }
&nbsp;
  handleChange(event) {
    let duration = event.target.value;
    this.setState({
      duration
    });
  }
&nbsp;
  activateAllZones(zones) {
    zones.forEach(zone =&gt; this.activateZone(zone.id));
  }
&nbsp;
  activateZone(id) {
    const duration = parseInt(this.state.duration);
    fetch('https://api.rach.io/1/public/zone/start', {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${apiKey}`
      },
      body: JSON.stringify({
        id,
        duration
      })
    })
      .then((response) =&gt; this.setState({
        status: [...this.state.status, response.status]
      }))
      .catch(<span class="fstat-no" title="function not covered" >er</span>ror =&gt; <span class="cstat-no" title="statement not covered" >this.setState({ message: `Error: ${error}` }))</span>;
  }
&nbsp;
  render() {
    let { device } = this.props;
    return (
      &lt;div className='device-card-container'&gt;
        &lt;div className='device-card'&gt;
          &lt;section className='device-card-text-section'&gt;
            &lt;h3 className='device-card-text'&gt;{device.name}&lt;/h3&gt;
            &lt;p className='device-card-text'&gt;Model: {device.model}&lt;/p&gt;
            &lt;p className='device-card-text'&gt;Status: {device.status}&lt;/p&gt;
            &lt;p className='device-card-text'&gt;Zones: {device.zones.length}&lt;/p&gt;
            &lt;button
              onClick={() =&gt; this.handleDisplayZones(device.zones)}
              className='buttons display-zones-button'
              value={device.id} &gt;
              {this.state.zones.length ? 'Hide Zones' : 'Display Zones'}
            &lt;/button&gt;
          &lt;/section&gt;
          &lt;section className='device-card-input-section'&gt;
            &lt;label
              htmlFor='duration'
              className='duration-label'&gt;
              Runtime Duration:
            &lt;/label&gt;
            &lt;input
              type='number'
              id='duration'
              className='duration-input'
              placeholder='Desired Runtime'
              value={this.state.duration}
              name='duration'
              onChange={(event) =&gt; this.handleChange(event)}&gt;
            &lt;/input&gt;
            &lt;button
              className='activate-buttons activate-all-button buttons'
              onClick={() =&gt; this.activateAllZones(device.zones)}&gt;
              Start All Zones
            &lt;/button&gt;
          &lt;/section&gt;
        &lt;/div&gt;
        &lt;div&gt;
          &lt;ZoneContainer 
            zones={this.state.zones} 
          /&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Oct 04 2018 19:19:04 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
