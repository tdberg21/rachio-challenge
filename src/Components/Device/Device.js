import React, { Component } from 'react';
import ZoneContainer from '../ZoneContainer/ZoneContainer.js';

export default class Device extends Component {
  constructor() {
    super();
    
    this.state = {
      zones: [],
      duration: 0
    };
  }

  handleDisplayZones(zones) {
    if (this.state.zones.length) {
      this.setState({
        zones: []
      });
    } else {
      let cleanZones = zones.map(zone => this.scrubZones(zone));
      this.setState({
        zones: cleanZones
      });
    }
  }

  scrubZones(zone) {
    let { id, name, zoneNumber, enabled, lastWateredDate, maxRuntime, imageUrl } = zone;
    let zoneInfo = {
      id,
      name,
      zoneNumber,
      enabled,
      lastWateredDate,
      maxRuntime,
      imageUrl
    };
    return zoneInfo;
  }

  handleChange(event) {
    let duration = event.target.value;
    this.setState({
      duration
    });
  }

  render() {
    let { device } = this.props;
    return (
      <div className='device-card'>
        <h3>{device.name}</h3>
        <p>Model: {device.model}</p>
        <p>Status: {device.status}</p>
        <p>Zones: {device.zones.length}</p>
        <input 
          type='number' 
          placeholder='Desired Runtime' 
          value={this.state.duration} 
          name='duration' onChange={(event) => this.handleChange(event)}>
        </input>
        <button 
          onClick={() => this.handleDisplayZones(device.zones)} 
          value={device.id} >
          {this.state.zones.length ? 'Hide Zones' : 'Display Zones'}
        </button>
        <button>Start All Zones</button>
        <div>
          <ZoneContainer zones={this.state.zones} duration={this.state.duration}/>
        </div>
      </div>
    );
  }
}