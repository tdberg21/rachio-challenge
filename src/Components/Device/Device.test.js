import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Device from './Device.js';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('Device tests', () => {
  let wrapper;
  let mockDevice = {
    id:'d8f8ac47',
    model: 'GENERATION2_8ZONE',
    name: 'Gouda',
    on: true,
    timeZone: 'America/Los_Angeles',
    zones: [{id: 1}, {id: 2}]
  };

  beforeEach(() => {
    wrapper = shallow(<Device device={mockDevice}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should clear the zones array in state when handleDisplayZones is called and there are zones in state', () => {
    const mockZones = [{ id: 1}, { id: 2}];
    wrapper = shallow(<Device device={mockDevice} />);
    wrapper.setState({zones: mockZones});
    wrapper.instance().handleDisplayZones();
    const results = wrapper.state('zones');

    expect(results).toEqual([]);
  });

  it('should add the zones to state when handleDisplayZones is called and there are no zones in state', () => {
    const mockZones = [{
      id: 1,
      name: 'taco',
      zoneNumber: 2,
      enabled: true,
      lastWateredDate: '06212018',
      maxRuntime: 10800,
      imageUrl: 'google.com'
    }, {
      id: 2,
      name: 'taco2',
      zoneNumber: 3,
      enabled: true,
      lastWateredDate: '06212018',
      maxRuntime: 10800,
      imageUrl: 'google.com'
    }];
    wrapper = shallow(<Device device={mockDevice} />);
    wrapper.setState({ zones: [] });
    wrapper.instance().handleDisplayZones(mockZones);
    const results = wrapper.state('zones').length;

    expect(results).toEqual(2);
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: { name: 'duration', value: 60 } };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('duration')).toBe(60);
  });

  it('should invoke handleChange when runtime duration is changed', () => {
    wrapper = shallow(<Device device={mockDevice} />);
    const spy = spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    const mockEvent = { target: { value: 'something' } };

    wrapper.find('.duration-input').simulate('change', mockEvent);

    expect(spy).toHaveBeenCalled();
  });

  it('should invoke handleDisplayZones on click of the display zones button', () => {
    const mockZones = [{ id: 1 }, { id: 2 }];
    wrapper = shallow(<Device device={mockDevice} />);
    wrapper.setState({ zones: mockZones });
    const spy = spyOn(wrapper.instance(), 'handleDisplayZones');
    wrapper.instance().forceUpdate();

    wrapper.find('.display-zones-button').simulate('click', );

    expect(spy).toHaveBeenCalled();
  });
});