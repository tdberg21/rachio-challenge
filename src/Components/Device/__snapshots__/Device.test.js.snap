// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device tests should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Device
    device={
      Object {
        "id": "d8f8ac47",
        "model": "GENERATION2_8ZONE",
        "name": "Gouda",
        "on": true,
        "timeZone": "America/Los_Angeles",
        "zones": Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="device-card-text-section"
        >
          <h3
            className="device-card-text"
          >
            Gouda
          </h3>
          <p
            className="device-card-text"
          >
            Model: 
            GENERATION2_8ZONE
          </p>
          <p
            className="device-card-text"
          >
            Status: 
          </p>
          <p
            className="device-card-text"
          >
            Zones: 
            2
          </p>
        </div>,
        <div
          className="device-card-input-section"
        >
          <button
            className="activate-buttons"
            onClick={[Function]}
          >
            Start All Zones
          </button>
          <label
            className="duration-label"
            htmlFor="duration"
          >
            Runtime Duration:
          </label>
          <input
            className="duration-input"
            id="duration"
            name="duration"
            onChange={[Function]}
            placeholder="Desired Runtime"
            type="number"
            value={0}
          />
          <button
            className="activate-buttons display-zones-button"
            onClick={[Function]}
            value="d8f8ac47"
          >
            Display Zones
          </button>
        </div>,
        <div>
          <ZoneContainer
            zones={Array []}
          />
        </div>,
      ],
      "className": "device-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="device-card-text"
            >
              Gouda
            </h3>,
            <p
              className="device-card-text"
            >
              Model: 
              GENERATION2_8ZONE
            </p>,
            <p
              className="device-card-text"
            >
              Status: 
            </p>,
            <p
              className="device-card-text"
            >
              Zones: 
              2
            </p>,
          ],
          "className": "device-card-text-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Gouda",
              "className": "device-card-text",
            },
            "ref": null,
            "rendered": "Gouda",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "GENERATION2_8ZONE",
              ],
              "className": "device-card-text",
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "GENERATION2_8ZONE",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Status: ",
                undefined,
              ],
              "className": "device-card-text",
            },
            "ref": null,
            "rendered": Array [
              "Status: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Zones: ",
                2,
              ],
              "className": "device-card-text",
            },
            "ref": null,
            "rendered": Array [
              "Zones: ",
              2,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="activate-buttons"
              onClick={[Function]}
            >
              Start All Zones
            </button>,
            <label
              className="duration-label"
              htmlFor="duration"
            >
              Runtime Duration:
            </label>,
            <input
              className="duration-input"
              id="duration"
              name="duration"
              onChange={[Function]}
              placeholder="Desired Runtime"
              type="number"
              value={0}
            />,
            <button
              className="activate-buttons display-zones-button"
              onClick={[Function]}
              value="d8f8ac47"
            >
              Display Zones
            </button>,
          ],
          "className": "device-card-input-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Start All Zones",
              "className": "activate-buttons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Start All Zones",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Runtime Duration:",
              "className": "duration-label",
              "htmlFor": "duration",
            },
            "ref": null,
            "rendered": "Runtime Duration:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "duration-input",
              "id": "duration",
              "name": "duration",
              "onChange": [Function],
              "placeholder": "Desired Runtime",
              "type": "number",
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Display Zones",
              "className": "activate-buttons display-zones-button",
              "onClick": [Function],
              "value": "d8f8ac47",
            },
            "ref": null,
            "rendered": "Display Zones",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ZoneContainer
            zones={Array []}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "zones": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="device-card-text-section"
          >
            <h3
              className="device-card-text"
            >
              Gouda
            </h3>
            <p
              className="device-card-text"
            >
              Model: 
              GENERATION2_8ZONE
            </p>
            <p
              className="device-card-text"
            >
              Status: 
            </p>
            <p
              className="device-card-text"
            >
              Zones: 
              2
            </p>
          </div>,
          <div
            className="device-card-input-section"
          >
            <button
              className="activate-buttons"
              onClick={[Function]}
            >
              Start All Zones
            </button>
            <label
              className="duration-label"
              htmlFor="duration"
            >
              Runtime Duration:
            </label>
            <input
              className="duration-input"
              id="duration"
              name="duration"
              onChange={[Function]}
              placeholder="Desired Runtime"
              type="number"
              value={0}
            />
            <button
              className="activate-buttons display-zones-button"
              onClick={[Function]}
              value="d8f8ac47"
            >
              Display Zones
            </button>
          </div>,
          <div>
            <ZoneContainer
              zones={Array []}
            />
          </div>,
        ],
        "className": "device-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="device-card-text"
              >
                Gouda
              </h3>,
              <p
                className="device-card-text"
              >
                Model: 
                GENERATION2_8ZONE
              </p>,
              <p
                className="device-card-text"
              >
                Status: 
              </p>,
              <p
                className="device-card-text"
              >
                Zones: 
                2
              </p>,
            ],
            "className": "device-card-text-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Gouda",
                "className": "device-card-text",
              },
              "ref": null,
              "rendered": "Gouda",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Model: ",
                  "GENERATION2_8ZONE",
                ],
                "className": "device-card-text",
              },
              "ref": null,
              "rendered": Array [
                "Model: ",
                "GENERATION2_8ZONE",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Status: ",
                  undefined,
                ],
                "className": "device-card-text",
              },
              "ref": null,
              "rendered": Array [
                "Status: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Zones: ",
                  2,
                ],
                "className": "device-card-text",
              },
              "ref": null,
              "rendered": Array [
                "Zones: ",
                2,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="activate-buttons"
                onClick={[Function]}
              >
                Start All Zones
              </button>,
              <label
                className="duration-label"
                htmlFor="duration"
              >
                Runtime Duration:
              </label>,
              <input
                className="duration-input"
                id="duration"
                name="duration"
                onChange={[Function]}
                placeholder="Desired Runtime"
                type="number"
                value={0}
              />,
              <button
                className="activate-buttons display-zones-button"
                onClick={[Function]}
                value="d8f8ac47"
              >
                Display Zones
              </button>,
            ],
            "className": "device-card-input-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Start All Zones",
                "className": "activate-buttons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Start All Zones",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Runtime Duration:",
                "className": "duration-label",
                "htmlFor": "duration",
              },
              "ref": null,
              "rendered": "Runtime Duration:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "duration-input",
                "id": "duration",
                "name": "duration",
                "onChange": [Function],
                "placeholder": "Desired Runtime",
                "type": "number",
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Display Zones",
                "className": "activate-buttons display-zones-button",
                "onClick": [Function],
                "value": "d8f8ac47",
              },
              "ref": null,
              "rendered": "Display Zones",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ZoneContainer
              zones={Array []}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "zones": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
